### ISSUE TITLE
pdf.image() doesn't allow file names starting with 'data'

### ISSUE THREAD

**Issue by 1nv** [PyFPDF/fpdf2#794]
**Error details**

When trying to call `pdf.image()` with `filename` starting with `'data'`, an error occurs:
```
imageData = base64Image.split("base64,")[1]
IndexError: list index out of range
```

This is caused by treating anything starting with `'data'` as base64 string (see `load_image()` with `elif filename.startswith("data"):` check), but some file paths might also start with those characters.

**Minimal code**

```python
from fpdf import FPDF

pdf = FPDF()
pdf.image('data/image.png', x=0, y=0, w=10, h=10)
```

**Environment**

* **Operating System**: Windows 10
* **Python version**: 3.10.11
* **`fpdf2` version used**: 2.7.4

**Comment by Lucas-C** [comment]
@allcontributors please add @1nv for bug

**Comment by 1nv** [comment]
Hi @Lucas-C,

> Would it be OK for you if we checked for the `data:` prefix instead of just data?

Yes, that's exactly what I was thinking of, since `:` is not a valid character for paths (except for drive letters on Windows and mount points on one other platform).

**Comment by Lucas-C** [comment]
> Yes, that's exactly what I was thinking of, since `:` is not a valid character for paths (except for drive letters on Windows and mount points on one other platform).

Thanks for your feedback

I merged #796

It is not released yet, but you can install the latest version of `fpdf2` straight from git this way:

**Comment by 1nv** [comment]
I just tested the way you said, and it works fine!
Thank you for the quick fix!
